xbmc/pvr/epg/EpgContainer.cpp

```diff
@@ -765,14 +765,15 @@ const CDateTime CPVREpgContainer::GetLastEPGDate(void)
 {
   CDateTime returnValue;
 
-  CSingleLock lock(m_critSection);
-  for (const auto &epgEntry : m_epgs)
+  m_critSection.lock();
```

Comment:
I think most of the files use this convention for locking a certain section only:

```
{
  CSingleLock lock(section);
  /* do work */
}
```

Reply:
Hehe, true and I actually had rewritten this to:

```C++
  std::map<unsigned int, CPVREpgPtr> epgs;
  {
    CSingleLock lock(m_critSection);
    epgs = m_epgs;
  }
````

 but suddenly felt the lock-do-unlock- sequnce more elegant. But I can change this if you like.

Reply:
I don't have a strong preference either way